source-code/sdk-js/src/adapter-sveltekit/ast-transformation/transforms/+layout.svelte.ts

```diff
@@ -9,76 +28,205 @@ export const transformLayoutSvelte = (config: TransformConfig, code: string, roo
+				b.blockStatement([addRuntimeToContextAst, initImportedVariablesAst]),
+			)
+			localLanguageName =
+				importNames.find(([imported]) => imported === "language")?.[1] ?? "language"
+			const reactiveLabeledStatementAst = b.labeledStatement(
+				b.identifier("$"),
+				b.ifStatement(
+					b.logicalExpression("&&", b.identifier("browser"), b.identifier("$" + localLanguageName)),
+					b.blockStatement([
+						b.expressionStatement(
+							b.chainExpression(
+								b.callExpression(
+									b.optionalMemberExpression(
+										b.memberExpression(
+											b.memberExpression(b.identifier("document"), b.identifier("body")),
+											b.identifier("parentElement"),
+										),
+										b.identifier("setAttribute"),
+									),
+									[b.literal("lang"), b.identifier("$" + localLanguageName)],
+								),
+							),
+						),
+						b.expressionStatement(
+							b.callExpression(
+								b.memberExpression(b.identifier("localStorage"), b.identifier("setItem")),
+								[b.identifier("localStorageKey"), b.identifier("$" + localLanguageName)],
+							),
+						),
+					]),
+				),
+			)
+			// Return the index in ast body that contains our first declaration
+			const usageIndexes = findAstJs(
+				ast.$ast,
+				[
+					({ node }) =>
+						n.Identifier.check(node) &&
+						(importNames.some(([, local]) => local === node.name) || node.name === "data"),
+				],
+				(node) =>
+					n.Identifier.check(node)
+						? (meta) => {
+								let { parent, index } = meta.get(node) ?? {}
+								while (parent != undefined && !n.Program.check(parent)) {
+									const parentMeta = meta.get(parent)
+									parent = parentMeta?.parent
+									index = parentMeta?.index
+								}
+								if (n.Program.check(parent)) return index
+								return undefined
+						  }
+						: undefined,
+			)[0] as number[] | undefined
+			// prefix language and i aliases with $ if reactive
+			if (!config.languageInUrl) makeJsReactive(ast.$ast, reactiveImportIdentifiers)
+			// Add exportLetDataExportAst, addRuntimeToContextAst and initImportedVariablesAst before any usage of "data" or importNames
+			if (n.Program.check(ast.$ast)) {
+				ast.$ast.body.splice(
+					usageIndexes?.[0] ?? ast.$ast.body.length,
+					0,
+					exportLetDataExportAst,
+					variableDeclarationAst,
+					addRuntimeToContextAst,
+					initImportedVariablesAst,
+					config.languageInUrl ? nonReactiveLabeledStatementAst : reactiveLabeledStatementAst,
+				)
+			}
 
-	addRuntimeToContext(getRuntimeFromData(data))
-	let { i, language } = getRuntimeFromContext()
-	${initCode}
-</script>
-${template}
-`
-}
+			const generated = generateCode(ast, {
+				sourceMapName: config.sourceMapName,
+			})
 
-// TODO: transform
-export const wrapRootLayoutSvelte = (config: TransformConfig, code: string) => {
-	// TODO: more meaningful error messages
-	throw new Error("currently not supported")
+			return { ...options, ...generated }
+		},
+		markup: (options) => {
```

Comment:
It seems you are doing the same things here that you do also in the `*.svelte.js`. We could just insert "regular" code here, and then pass it to `transformSvelte`. Should make the implementation easier

Reply:
Hmm, I think sadly we can't because it will destroy the sourcemapping as `preprocess` doesn't take a sourcemap as input

Reply:
we don't care for now. We can figure it out later